elytron/src/main/java/org/wildfly/extension/elytron/PolicyDefinitions.java

```diff
@@ -237,6 +239,15 @@ private Policy getPolicy() {
                 ElytronExtension.getResourceDescriptionResolver(POLICY))
                 .setAddHandler(add)
                 .setRemoveHandler(new ReloadRequiredRemoveStepHandler() {
+                    @Override
+                    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+                        super.performRuntime(context, operation, model);
+                        // JACC settings require a restart to take place
+                        if (model.get(JACC_POLICY).isDefined()) {
+                            context.restartRequired();
+                        }
+                    }
+
```

Comment:
This needs to override recoverServices as well.

I think it would be better if this were just an AbstractRemoveStepHandler, not ReloadRequiredRemoveStepHandler, and then this  explicitly deals with the reload-required case in performRuntime and recoverServices. Calling super.performRuntime(context, operation, model) to deal with the non-restart-required case obscures what is really happening.